frontend/src/components/window/Tab.js

```diff
@@ -31,16 +35,17 @@ class Tab extends Component {
         : '';
 
       if (singleRowView) {
-        getTab(tabId, windowId, docId).then((res) => {
+        fetchTab(tabId, windowId, docId).then((res) => {
           if (res.length) {
-            dispatch(updateMasterData(res[0]));
-            dispatch(addRowData({ [tabId]: res }, 'master'));
+            updateMasterData(res[0]);
+            addRowData({ [tabId]: res }, 'master');
             onChange && onChange();
           }
         });
+        fetchTab(tabId, windowId, docId);
       } else {
-        getTab(tabId, windowId, docId, query).then((res) => {
-          dispatch(addRowData({ [tabId]: res }, 'master'));
+        fetchTab(tabId, windowId, docId, query).then((res) => {
```

Comment:
I don't know who Ryan McDermott is so not really convinced his style guide is any kind of market standard. As mentioned above - although it does make sense to switch to named parameters in some cases, most of the time it's a premature optimization. And in our case we're not great at naming conventions, and often we use different names for the same parameter (`windowId/windowType`) or have duplicated names (`docId/docid`). So there might be unexpected side effects.

Reply:
> I don't know who Ryan McDermott 

me neither but those are the coding standards we (all) agreed to use.
Ofc we can improve them but i am eager to understand why when having a certain number of params, why is not using named parameters?

PS: that coding style was adapted from Uncle Bob's Clean Code book.... which is widely adopted.

> And in our case we're not great at naming conventions, 

That's true.

> and often we use different names for the same parameter (windowId/windowType) or have duplicated names (docId/docid).

That's also true.

> So there might be unexpected side effects.

can u pls enumarete a couple of those side effects?

Reply:
Like... shit breaks because we're not sending the right parameter ? `windowId` is undefined in the URL ?

Reply:
can u give me a concrete example? i.e. which functions might break?

I briefly scrolled over the code and I saw those param names are quite local.
Also pressing F2 in VSCode will do the rename for you without any trouble.

E separate task would mean separate testing.
And a task like 'refactor the var name' i barely can see approved.

I could understand refactoring in interactions and have different iterations because you want to refactor & stabilize  (btw, Martin Fowler has a nice book about Refactoring techniques),
but it's hard for me to see a refactor var names task realistic.

I know that we were not good at naming things, so we named the same thing in several ways introducing to developers.
But we can change. We can fix it one by one when we touch that part of the code.
Else, my fear is that in one year we will talk about same problems and we will contemplate same issues.